internal/gatewayapi/listener.go

```diff
@@ -188,24 +188,28 @@ func (t *Translator) processProxyObservability(gwCtx *GatewayContext, xdsIR *ir.
 
 	xdsIR.AccessLog, err = t.processAccessLog(envoyProxy, resources)
 	if err != nil {
-		status.UpdateGatewayListenersNotValidCondition(gwCtx.Gateway, gwapiv1.GatewayReasonInvalid, metav1.ConditionFalse,
+		status.SetGatewayAccepted(gwCtx.Gateway, false, gwapiv1.GatewayReasonInvalidParameters,
 			fmt.Sprintf("Invalid access log backendRefs: %v", err))
 		return
 	}
 
 	xdsIR.Tracing, err = t.processTracing(gwCtx.Gateway, envoyProxy, t.MergeGateways, resources)
 	if err != nil {
-		status.UpdateGatewayListenersNotValidCondition(gwCtx.Gateway, gwapiv1.GatewayReasonInvalid, metav1.ConditionFalse,
+		status.SetGatewayAccepted(gwCtx.Gateway, false, gwapiv1.GatewayReasonInvalidParameters,
 			fmt.Sprintf("Invalid tracing backendRefs: %v", err))
 		return
 	}
 
 	xdsIR.Metrics, err = t.processMetrics(envoyProxy, resources)
 	if err != nil {
-		status.UpdateGatewayListenersNotValidCondition(gwCtx.Gateway, gwapiv1.GatewayReasonInvalid, metav1.ConditionFalse,
+		status.SetGatewayAccepted(gwCtx.Gateway, false, gwapiv1.GatewayReasonInvalidParameters,
 			fmt.Sprintf("Invalid metrics backendRefs: %v", err))
 		return
 	}
+	
+	// need explicit setting accepted status to true, as the status may have been set to false in previous loop,
+	// e.g. when access log backendRefs created later than the Gateway or EnvoyProxy.
+	status.SetGatewayAccepted(gwCtx.Gateway, true, gwapiv1.GatewayReasonAccepted, "The Gateway has not been scheduled by Envoy Gateway")
```

Comment:
there are a few issues here
* we've added a new status condition that did not apply before
* accepted=true but the message says `The Gateway has not been scheduled by Envoy Gateway` , this is not useful for the user

Reply:
my bad, it should be `The Gateway has not been scheduled by Envoy Gateway`.